ui/src/app/machines/views/MachineList/StatusColumn/StatusColumn.js

```diff
@@ -0,0 +1,131 @@
+import { useSelector } from "react-redux";
+import React from "react";
+import PropTypes from "prop-types";
+
+import {
+  general as generalSelectors,
+  machine as machineSelectors
+} from "app/base/selectors";
+import { nodeStatus, scriptStatus } from "app/base/enum";
+import Tooltip from "app/base/components/Tooltip";
+
+// Node statuses for which the failed test warning is not shown.
+const hideFailedTestWarningStatuses = [
+  nodeStatus.COMMISSIONING,
+  nodeStatus.FAILED_COMMISSIONING,
+  nodeStatus.FAILED_TESTING,
+  nodeStatus.NEW,
+  nodeStatus.TESTING
+];
+
+// Node statuses for which the OS + release is made human-readable.
+const formattedReleaseStatuses = [nodeStatus.DEPLOYED, nodeStatus.DEPLOYING];
+
+// Node statuses that are temporary.
+const transientStatuses = [
+  nodeStatus.COMMISSIONING,
+  nodeStatus.DEPLOYING,
+  nodeStatus.DISK_ERASING,
+  nodeStatus.ENTERING_RESCUE_MODE,
+  nodeStatus.EXITING_RESCUE_MODE,
+  nodeStatus.RELEASING,
+  nodeStatus.TESTING
+];
+
+// Script statuses associated with failure.
+const failedScriptStatuses = [
+  scriptStatus.DEGRADED,
+  scriptStatus.FAILED,
+  scriptStatus.FAILED_APPLYING_NETCONF,
+  scriptStatus.FAILED_INSTALLING,
+  scriptStatus.TIMEDOUT
+];
+
+const getStatusText = (machine, osReleases) => {
+  if (formattedReleaseStatuses.includes(machine.status_code)) {
+    const machineRelease = osReleases.find(
+      release => release.value === machine.distro_series
+    );
+
+    if (machineRelease) {
+      let releaseTitle;
+      if (machine.osystem === "ubuntu") {
+        releaseTitle = machineRelease.label.split('"')[0].trim();
+      } else {
+        releaseTitle = machineRelease.label;
+      }
+
+      if (machine.status_code === nodeStatus.DEPLOYING) {
+        return `Deploying ${releaseTitle}`;
+      }
+      return releaseTitle;
+    }
+  }
+  return machine.status;
+};
+
+const getProgressText = machine => {
+  if (transientStatuses.includes(machine.status_code)) {
+    return machine.status_message;
+  }
+  return "";
+};
+
+const getStatusIcon = machine => {
+  if (transientStatuses.includes(machine.status_code)) {
+    return (
+      <i
+        className="p-icon--spinner u-animation--spin"
+        data-test="status-icon"
+      />
```

Comment:
Could we use the Loading component for this?

Reply:
@huwshimi Updated. I've had to add "u-no-padding u-no-margin" to the Loader otherwise it popped out of its container - maybe we should add an issue in react-components?